tools/cfn2ts/lib/codegen.ts

```diff
@@ -514,15 +538,29 @@ export default class CodeGenerator {
     this.code.closeBlock();
   }
 
-  private emitProperty(context: genspec.CodeName, propName: string, spec: schema.Property, additionalDocs: string): string {
-    const question = spec.Required ? '' : '?';
-    const javascriptPropertyName = genspec.cloudFormationToScriptName(propName);
+  private emitProperty(props: EmitPropertyProps): string {
+    const javascriptPropertyName = genspec.cloudFormationToScriptName(props.propName);
 
-    this.docLink(spec.Documentation, additionalDocs);
-    this.code.line(`readonly ${javascriptPropertyName}${question}: ${this.findNativeType(context, spec, propName)};`);
+    this.docLink(props.spec.Documentation, props.additionalDocs);
+    const line = `: ${this.findNativeType(props.context, props.spec, props.propName)};`;
 
+    if (props.container === Container.Interface) {
```

Comment:
TBH not sure there is too much value in making this a single method with two very different code branches. It would be more readable to just implement this as two separate methods....

Reply:
done - split into two functions.

Reply:
👍